node/core/approval-voting/src/import.rs

```diff
@@ -360,25 +361,7 @@ pub(crate) async fn handle_new_head<Context, B: Backend>(
 	};
 
 	// Update session info based on most recent head.
-	match state.cache_session_info_for_head(ctx, head).await {
-		Err(e) => {
-			gum::debug!(
-				target: LOG_TARGET,
-				?head,
-				?e,
-				"Could not cache session info when processing head.",
-			);
-			return Ok(Vec::new())
-		},
-		Ok(Some(a @ SessionWindowUpdate::Advanced { .. })) => {
-			gum::info!(
-				target: LOG_TARGET,
-				update = ?a,
-				"Advanced session window for approvals",
-			);
-		},
-		Ok(_) => {},
-	}
+	cache_session_info_for_head(ctx.sender(), head, session_info_provider).await;
```

Comment:
if I'm reading this correctly, this will update the highest_seen_session and lead to bailing on ancient session error in case finality stalls for 6+ sessions?

Reply:
Yes, but this is just caching. If the session is actually needed - it will be queried with a relay parent and should be obtainable.

Reply:
I've removed the caching altogether.